tuf/data/roles.go

```diff
@@ -96,6 +97,16 @@ func IsBaseRole(role string) bool {
 	return false
 }
 
+// IsWildDelegation determines if a role represents a valid wildcard delegation
+// path, i.e. targets/*, targets/foo/*.
+// The wildcard may only appear as the final part of the delegation and must
+// be a whole segment, i.e. targets/foo* is not a valid wildcard delegation.
+func IsWildDelegation(role string) bool {
+	base := filepath.Dir(role)
```

Comment:
Oop, looks like the `path.Dir` will then count the `*` as part of the dir, if there are trailing slashes, so hopefully `IsDelegation` will fail that one.  But possibly slashes in the middle might be ignored?  https://play.golang.org/p/ChBOllvy42